### PR TITLE
correctly validate whether response selector responses from stories are valid if responses are found in the domain

### PR DESCRIPTION
**Pull request by erohmensing**
**Proposed changes**:
- Only recognize the first chunk of response selector responses when looking for names of actions in the domain file for action/story validation (e.g. `utter_chitchat` of `utter_chitchat/ask_name`
- Fix https://github.com/RasaHQ/rasa/issues/8633

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: data/test_domains/default_retrieval_intents.yml [hunk 1]
```diff
@@ -22,12 +22,3 @@ responses:
...
     - text: I am a bot, powered by Rasa.
-
-actions:
-  - utter_chitchat
-  - utter_greet
-  - utter_cheer_up
-  - utter_did_that_help
-  - utter_happy
-  - utter_goodbye
-  - utter_iamabot
```
Comment (ancalita) [inline data/test_domains/default_retrieval_intents.yml:33]: Why were the actions removed in this file?

**Review by ancalita — COMMENTED** [review]
The changes look great! One tiny thing though - could you please add the context behind this fix and/or the GH issue link in the PR description, it's not very clear until you read the changelog.

**Comment by erohmensing** [comment]
Oh whoops! I didn't realize i forgot to update it, thanks!

File: data/test_domains/default_retrieval_intents.yml (hunk 1 above)
Comment (erohmensing) [reply data/test_domains/default_retrieval_intents.yml:33]: We removed the requirement to list the reponses as actions a long time ago. In copying it over to the `data/test_domains/response_selector_responses_in_domain.yml`, it made sense to test that the `utter_chitchat` action counted as a valid utterance action after the fix even if you didn't list it under `actions`, so I didn't want to copy it over in that case -- to made more sense to remove it from the original one, but i can also put it back, shouldn't affect it either way 😄

**Review by ancalita — APPROVED** [review]

**Pull request merged**